test/common/index.js

```diff
@@ -616,6 +616,19 @@ function expectsError(fn, settings, exact) {
   return mustCall(innerFn, exact);
 }
 
+const suffix = 'This is caused by either a bug in Node.js ' +
+  'or incorrect usage of Node.js internals.\n' +
+  'Please open an issue with this stack trace at ' +
+  'https://github.com/nodejs/node/issues\n';
+
+function expectsInternalAssertion(fn, message) {
+  assert.throws(fn, {
+    message: `${message}\n${suffix}`,
+    name: 'Error [ERR_INTERNAL_ASSERTION]',
+    code: 'ERR_INTERNAL_ASSERTION'
+  });
+}
```

Comment:
I would rather not add this helper function in common. Common is AFAIC pretty overboarded.

Reply:
That's where `common.expectsError` is...I guess we could move these and other `expectsSomething` to one file when the time comes

Reply:
Also, surprisingly, `test/common.js` is even less than 1000 lines? That's even smaller than a lot of our tests :S

Reply:
It's not about having a lot of lines but this file is loaded for every single test file and I would rather completely get rid of something that we always use.

Having something more specific would be fine but I actually doubt that this will often be tested for and currently there are only for places in which this helper would be used in.

About `common.expectsError()` it is already used significantly less as we can always use `assert.throws()` instead. Only callbacks should always use `common.expectsError()`.

Reply:
Do you strongly prefer the helper function? To me it still does not seem necessary.

Reply:
@BridgeAR Yes, I prefer using it instead of repeating the string around the test. That's what `test/common` is intended for - to DRY the tests.

I don't really see the need to shrink `test/common`, TBH. If the concern does not come from the size of the file but instead of time spent loading it, then we should try creating code cache for it in the test runner instead of limiting ourselves. For tests that run without flags (or just with --expose-internals) we could even get the test runner create snapshot first and load it to run the tests because at that point all environment states are stable and controlled (that's also what V8 does in its infra).

Reply:
I believe the consensus is you don't need to validate the `message` of errors in more then one place (in this case `test/message/internal_assert.js` is enough), so there's no need for the value that you are DRYing in the first place.

Reply:
@refack This is special in that the message is what we want to test for in the test cases, because the code is constant across all internal assertions and we intentionally do not give error codes for assertions coming out of completely different type of bugs.

For instance, we want to verify that we tell the user `handle must be a FSEvent` if the user monkey patches `watcher._handle`, but we do not want to create a code for that special type of bugs because the user is simply violating the API contract and it gives us little to classify all these violations other than just telling them no.

Reply:
I think it's important to make sure we provide useful information in the assertions, even for our own sake - just saying that "you are hitting a bug, please open an issue" is not enough, see https://github.com/nodejs/node/pull/26859 (though that one is not verifiable in user land).

Reply:
> This is special in that the message is what we want to test for in the test cases,

What I mean there's no need to validate the message formatting mechanism more then once.
For example you can store the specific prefix behind a `Symbol` field and validate just it.

Reply:
My concern about `common/index` is only about the requirement of a common helper function. I would love to remove the requirement for our test files at some point.

I already gave my LG, so from my side this can land as is.